js_modules/dagit/packages/core/src/assets/LaunchAssetChoosePartitionsDialog.tsx

```diff
@@ -681,3 +684,103 @@ export const LAUNCH_ASSET_CHOOSE_PARTITIONS_QUERY = gql`
   ${DAEMON_NOT_RUNNING_ALERT_INSTANCE_FRAGMENT}
   ${USING_DEFAULT_LAUNCH_ERALERT_INSTANCE_FRAGMENT}
 `;
+
+const Warnings = ({
+  isPureAssetBackfill,
+  launchAsBackfill,
+  upstreamAssetKeys,
+  selections,
+  setSelections,
+}: {
+  launchAsBackfill: boolean;
+  isPureAssetBackfill: boolean;
+  upstreamAssetKeys: AssetKey[];
+  selections: PartitionDimensionSelection[];
+  setSelections: (next: PartitionDimensionSelection[]) => void;
+}) => {
+  const instanceResult = useQuery<
+    LaunchAssetChoosePartitionsQuery,
+    LaunchAssetChoosePartitionsQueryVariables
+  >(LAUNCH_ASSET_CHOOSE_PARTITIONS_QUERY);
+  const instance = instanceResult.data?.instance;
+
+  const alerts = [
+    isPureAssetBackfill ? (
+      <Alert
+        key="alert"
+        intent="info"
+        title="Dagster will materialize all partitions downstream of the selected partitions for the selected assets, using separate runs as needed."
+      />
+    ) : null,
+    UpstreamUnavailableWarning({
```

Comment:
This is pretty hacky because this component uses a hook so it shouldn't be used like a component but since I'm not calling it conditionally I'm basically using the component like a hook and it's fine, no hook rules broken... still super cringe though... but it beats the alternative of moving all of the logic up to this component (imo)

Reply:
> has it's data cached it would never render for a first time

It would still render. The hooks being used by that component get attributed to the Warnings component which means the Warnings component will re-render and recall that component again. Technically, the only difference is that there is no React element boundary but practically it works exactly the same.